### DISCUSSION TITLE
gfm in pandoc v2.9.2.1 causes PDF rendering issues related to markdown file

### DISCUSSION THREAD

**Discussion by skpaz** [jgm/pandoc#10455] [Q&A]
- Ubuntu 22.04.5 LTS 
- pandoc v2.9.2.1 (via apt)

pandoc v2.9.2 on with `--from='gfm'` set causes some PDF rendering issues. I do not experience the same issue with pandoc 3.5.1.

Example command:

```
pandoc \
--from gfm+emoji \
--toc \
--toc-depth 2 \
-V 'toc-title=Table of Contents' '--pdf-engine=weasyprint' '--pdf-engine-opt=--optimize-images' '--pdf-engine-opt=--debug' \
--css /home/runner/work/best-practice-guides/best-practice-guides/tools/../assets/css/book.css \
--css /home/runner/work/best-practice-guides/best-practice-guides/tools/../assets/css/github-markdown.css \
--css https://REDACTED.com/web/fonts/inter.css \
--quiet \
--verbose \
--strip-comments -o /home/runner/work/best-practice-guides/best-practice-guides/build/internal/writing-guides/writing-guides-pdf-markdown-content.pdf /home/runner/work/best-practice-guides/best-practice-guides/guides/internal/writing-guides/pdf-markdown/00-metadata.md /home/runner/work/best-practice-guides/best-practice-guides/guides/internal/writing-guides/pdf-markdown/01-introduction.md /home/runner/work/best-practice-guides/best-practice-guides/guides/internal/writing-guides/pdf-markdown/02-admonitions.md
```

I see this error, which does not happen with `--from=markdown` on v2.9.2.1 or at all on v3.5.

```
[INFO] No value for 'lang' was specified in the metadata.
  It is recommended that lang be specified for this format.
Warning:  This document format requires a nonempty <title> element.
  Defaulting to '00-metadata' as the title.
  To specify a title, use 'title' in metadata or --metadata title="...".
```

This is the same error we receive if the `00-metadata.md` file is not present, suggesting `--from='gfm'` has issues parsing the metdata file. 

When this happens on v2.9.2.1, the contents of the metadata file are instead rendered as HTML.

Example:

```html
<p>title: PDF Markdown author:</p>
<ul>
<li>AUTHOR_NAME_HERE lang: en-US book: true date: 2024-12-11 rights: © 2024 AUTHOR_NAME_HERE</li>
</ul>
<hr />
<!-- leave this blank -->
```

Here are the contents of `00-metadata.md`:

```yaml
---
title: PDF Markdown
author:
  -   AUTHOR_NAME_HERE
lang: en-US
book: true
date: 2024-12-11
rights: © 2024 AUTHOR_NAME_HERE
---
<!-- leave this blank -->
```

Is there a way to work around this issue on 2.9.2.1 with gfm, or is the only/best recourse to update to 3.5+?

v3.5+ would require a manual install, which isn't a huge problem, but a fix/workaround would require less testing on our side.

**Comment by jgm** [comment]
We don't backport fixes to earlier versions; just upgrade.
(Btw, when pandoc 2.9.2.1 was released, gfm didn't have any special handling for this format of metadata.)

**Comment by skpaz** [comment]
I wasn't concerned about a backport, I was just curious about the behavior. Ultimately, we manually installed pandoc 3.6 instead of via the distro's package repo. That solved the rendering issue.
